lib/auth/session_access.go

```diff
@@ -286,6 +286,17 @@ policySetLoop:
 			continue
 		}
 
+		if options.OnLeaveAction != types.OnSessionLeaveTerminate {
```

Comment:
This code assumes there are only two states, but the `OnSessionLeaveAction` type is able to accept future additional states.  Although I can't predict what we may in the future this may be a potential bug if we do add another leave action.

We could switch from a string to just a boolean to represent true to one of the possible exit states.

Or switching this to `options.OnLeaveAction == types.OnSessionLeavePause` may be incrementally safer.